addons/mail/static/src/messaging/service/messaging_service/messaging_service.js

```diff
@@ -34,6 +39,15 @@ const MessagingService = AbstractService.extend({
             // TODO FIXME The method uses service specific env keys so it can
             // only be called after a service has properly set up those keys.
             await this.env.messaging.start();
+        }).catch(error => {
+            if (this.env.shouldRaiseEntityDeletedError) {
+                throw error;
+            }
+            if (error instanceof EntityDeletedError) {
+                // Ignore entity deleted error.
+                return;
+            }
+            throw error;
```

Comment:
The 2 conditions need to be combined, `shouldRaiseEntityDeletedError` doesn't make sense outside of the error being the deleted type.